samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -113,9 +113,11 @@ void BLEConnectivityManager::DisconnectionHandler(bt_conn *conn, uint8_t reason)
 	/* Try to find provider matching the connection */
 	BLEBridgedDeviceProvider *provider = Instance().FindBLEProvider(*bt_conn_get_dst(conn));
 
-	bt_conn_unref(conn);
-
 	if (provider) {
+		if (conn) {
```

Comment:
`conn` is never null per API contract, right? BTW, is it correct to unref the connection in the disconnection handler? Per documentation, when the handler is called, there might be only one reference to the connection object left. So by unref you effectively destroy the object. Or did you `bt_conn_ref` this connection object somewhere?

Reply:
Right, removed `conn` check. As discussed F2F calling unref seems to be necessary by the `bt_conn_le_create` method API. I'm still not 100% sure if we do that correctly, however it's the only way that functionally works :) If we do not invoke unref we are not able to create new connection after it was closed (connect->disconnect->connect again scenario).